src/test/scala/io/iohk/ethereum/db/storage/StateStorageSpec.scala

```diff
@@ -0,0 +1,168 @@
+package io.iohk.ethereum.db.storage
+
+import akka.util.ByteString
+import io.iohk.ethereum.ObjectGenerators
+import io.iohk.ethereum.db.cache.MapCache
+import io.iohk.ethereum.db.dataSource.EphemDataSource
+import io.iohk.ethereum.db.storage.NodeStorage.{NodeEncoded, NodeHash}
+import io.iohk.ethereum.db.storage.pruning.{ArchivePruning, BasicPruning}
+import org.scalatest.{FlatSpec, Matchers}
+import org.scalatest.prop.PropertyChecks
+
+class StateStorageSpec extends FlatSpec with Matchers with PropertyChecks with ObjectGenerators {
+
+  "ArchiveStateStorage" should "save node directly to db" in new TestSetup {
+    forAll(keyValueByteStringGen(32)) { keyvals =>
+      keyvals.foreach { case (key, value) =>
+        archiveStateStorage.saveNode(key, value , 10)
+      }
+
+      keyvals.foreach { case (key, value) =>
```

Comment:
why not put it in one `foreach`?

Reply:
I wanted to first save all elements and then check than all of those elements are present. Puting it all in one `foreach` would be saving and checking one by one.  Not a big difference but i would leave it as it is.